frontend/public/components/import-yaml-results.tsx

```diff
@@ -0,0 +1,230 @@
+import { useTranslation } from 'react-i18next';
+import { Map as ImmutableMap } from 'immutable';
+import { Bullseye, Button, Spinner } from '@patternfly/react-core';
+import {
+  GreenCheckCircleIcon,
+  RedExclamationCircleIcon,
+  YellowExclamationTriangleIcon,
+} from '@console/shared';
+import {
+  TableComposable,
+  TableGridBreakpoint,
+  Tbody,
+  Td,
+  Th,
+  Thead,
+  Tr,
+} from '@patternfly/react-table';
+
+import { ResourceLink } from './utils';
+import { createResources } from './edit-yaml';
+import { K8sKind, referenceFor } from '../module/k8s';
+
+/**
+ * Without this prop our current TS types fail to match and require a `translate` prop to be added. PF suggests we
+ * update our types, but that causes other issues. This will have to do as a workaround for now.
+ *
+ * This is the best that I can find relates to this value:
+ * https://github.com/DefinitelyTyped/DefinitelyTyped/blob/3423b4fc3e3da09f8acc386bc2fee6fb8f5e0880/types/react/index.d.ts#L1763
+ */
+const reactPropFix = {
+  translate: 'no',
+};
+
+const ImportYAMLPageStatus: React.FC<ImportYAMLStatusProps> = ({ errors, inFlight }) => {
+  const { t } = useTranslation();
+  let StatusBlock: React.ReactNode;
+
+  if (inFlight) {
+    StatusBlock = (
+      <>
+        <Spinner size="lg" />
+        <h2>{t('public~Creating resources...')}</h2>
+      </>
+    );
+  } else if (!inFlight && !errors) {
+    StatusBlock = (
+      <>
+        <GreenCheckCircleIcon size="lg" />
+        <h2>{t('public~Resources successfully created')}</h2>
+      </>
+    );
+  } else {
+    StatusBlock = (
+      <>
+        <YellowExclamationTriangleIcon size="lg" />
+        <h2>{t('public~One or more resources failed to be created')}</h2>
+      </>
+    );
+  }
+  return <div className="co-import-yaml-status text-center">{StatusBlock}</div>;
+};
+
+const ImportYAMLResourceStatus: React.FC<ImportYAMLResourceStatusProps> = ({
+  creating,
+  error,
+  message,
+}) => {
+  let StatusIcon: React.ReactNode;
+  if (creating) {
+    StatusIcon = <Spinner size="sm" className="co-icon-space-r" />;
+  } else if (error) {
+    StatusIcon = <RedExclamationCircleIcon className="co-icon-space-r" size="sm" />;
+  } else {
+    StatusIcon = <GreenCheckCircleIcon className="co-icon-space-r" size="sm" />;
+  }
+  return (
+    <span>
+      {StatusIcon}
+      {message}
+    </span>
+  );
+};
+
+export const ImportYAMLResults: React.FC<ImportYAMLResultsProps> = ({
+  displayResults,
+  importResources,
+  models,
+  retryFailed,
+}) => {
+  const { t } = useTranslation();
+  const [importStatus, setImportStatus] = React.useState<ImportYAMLResourceStatusProps[]>(
+    importResources.map(() => ({
+      creating: true,
+      message: t('public~Creating'),
+    })),
+  );
+  const [inFlight, setInFlight] = React.useState(true);
+  const [errors, setErrors] = React.useState(false);
+  React.useEffect(() => {
+    const resources = createResources(importResources, models);
```

Comment:
These aren't resources... The name is misleading since they are promises for the HTTP requests. I would call this either `requests` or `promises`.

Reply:
changed this to requests in stead of resources.